sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedParquetRecordReader.java

```diff
@@ -154,8 +159,6 @@ public void close() throws IOException {
 
   @Override
   public boolean nextKeyValue() throws IOException {
-    resultBatch();
```

Comment:
I like this direction to reduce overhead. I confirmed that the current code calls `initBatch()` before calling `nextKeyValue()`, too.  
My question is that is calling sequence is mandatory or optional. In the future, I wonder a developer may call this method thru `hasNext()` before calling `initBatch()`.

How about moving `if-statement` into here like? Does this help performance improvement?
```
if (columnarBatch == null) resultBatch();
```
I am curious how orc reader handle this.
cc @dongjoon-hyun @cloud-fan

Reply:
Thanks for your review.
calling sequence is optional. `initBatch()` was moved to `nextBatch()`, if `hasNext()` called before `initBatch()`, eventually `initBatch()` will be called first of all in `nextBatch()`:
```
  public boolean nextBatch() throws IOException {
     if (columnarBatch == null) initBatch();
```

Orc has a much simple way: 
```
  public boolean nextKeyValue() throws IOException {
    return nextBatch();
  }
```